version: "3.7"

volumes:
  pgdata:
  redisdata:

networks:
  private:

services:
  redis:
    image: redis:alpine
    restart: unless-stopped
    volumes:
      - redisdata:/data
    networks:
      - private

  postgres:
    image: postgres:alpine
    restart: unless-stopped
    env_file:
      - .env
    volumes:
      - pgdata:/var/lib/postgresql/data
    networks:
      - private

  backend:
    image: ghcr.io/artrey/tinkoff-monitor:latest
    restart: unless-stopped
    env_file:
      - .env
    depends_on:
      - postgres
      - redis
    volumes:
      - ./static:/app/static
      - ./media:/app/media
    networks:
      - private

  celery:
    image: ghcr.io/artrey/tinkoff-monitor:latest
    restart: unless-stopped
    env_file:
      - .env
    depends_on:
      - postgres
      - redis
    volumes:
      - ./static:/app/static:ro
      - ./media:/app/media:ro
    networks:
      - private
    command: "celery -A tinkoff worker --loglevel error --pool=solo"

#  beat:
#    image: ghcr.io/artrey/tinkoff-monitor:latest
#    restart: unless-stopped
#    env_file:
#      - .env
#    depends_on:
#      - postgres
#      - redis
#    networks:
#      - private
#    command: "celery -A tinkoff beat"

#  bot:
#    image: ghcr.io/artrey/tinkoff-monitor:latest
#    restart: unless-stopped
#    env_file:
#      - .env
#    depends_on:
#      - postgres
#      - redis
#    networks:
#      - private
#    command: "python manage.py start_bot"

  nginx:
    image: nginx:stable-alpine
    restart: unless-stopped
    depends_on:
      - backend
    volumes:
      - ./static:/var/www/django/static:ro
      - ./media:/var/www/django/media:ro
      - ./nginx.conf:/etc/nginx/conf.d/default.conf:ro
      - ./logs:/var/log/nginx
    networks:
      - private
    ports:
      - 80:80
